### ISSUE TITLE
[HOLD] [$2000] Firefox - the wrong chat DM is opening when attempting to search/DM a specific user

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#10884]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Opened my test account in firefox web browser 
2. Typed cmd+k to open search/chat switcher
3. Typed [mallen@expensify.com](mailto:mallen@expensify.com) to send my main account a message
4. Clicked on the returned account for [mallen@expensify.com](mailto:mallen@expensify.com) (also tested by hitting 'return')

## Expected Result:
My [mallen@expensify.com](mailto:mallen@expensify.com) account to open in the main chat window

## Actual Result:
Another test account ([matt+4499384dhfj@icecreamman.com](mailto:matt+4499384dhfj@icecreamman.com)) opened in main chat window

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web - Firefox

**Version Number:** 1.1.92-0
**Reproducible in staging?:** Y
**Reproducible in production?:** 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 

https://user-images.githubusercontent.com/43995119/189031449-b398b97b-c35a-41a7-84b5-61538012f3b2.mp4

[opening wrong chat.txt](https://github.com/Expensify/App/files/9523456/opening.wrong.chat.txt)

**Upwork job URL:** https://www.upwork.com/jobs/~0114fee954d1331354
**Issue reported by:** @mallenexpensify 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1661799578984039

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mallenexpensify** [comment]
woah... missed this.  Let me try to repro today/tomorrow

**Comment by mallenexpensify** [comment]
Able to reproduce on Web-Firefox, moving to on to engineering

https://user-images.githubusercontent.com/22508304/192607740-1f93bd98-176f-4b27-857c-c8390bff1989.mp4

Logs
[wrong chat opening.txt](https://github.com/Expensify/App/files/9658744/wrong.chat.opening.txt)

**Comment by michaelhaxhiu** [comment]
Reproduced, and boy is this weird.
![weirdaf](https://user-images.githubusercontent.com/13206896/193600105-50af842a-0cf1-438d-beef-b3c37997265a.gif)

Notice how it fails the first time, but not the second time? Also, it doesn't work consistently for all users (like I tried with lauren@expensify.com and employee.mixalis@expensify.com and it worked fine!)

https://recordit.co/WvrWKEIQwX

**Comment by michaelhaxhiu** [comment]
Note - I am posting this to Upwork as I'm fairly certain it's `external` @iwiznia.

**Comment by michaelhaxhiu** [comment]
Not overdue yet, still fielding proposals.

**Comment by michaelhaxhiu** [comment]
Doubling price!

**Comment by michaelhaxhiu** [comment]
Doubling

**Comment by michaelhaxhiu** [comment]
Yep, it was on my list. I'll double it now since I got the ping from ya :)

**Comment by vladamx** [comment]
**Proposal**

@parasharrajat This is apparently fixing the issue.

<img width="1057" alt="image" src="https://user-images.githubusercontent.com/11381171/197899130-281f11ec-2ee8-4a2f-bb11-633379850439.png">

**Comment by parasharrajat** [comment]
Sorry, I am late to mention this but this issue should be on hold. we(I) know the reason for this issue and a proper solution is possible only after we are done with the navigation redesign. 

There is a tracking issue for the same. https://github.com/Expensify/App/issues/11768. 


cc: @michaelhaxhiu @rushatgabhane

**Comment by michaelhaxhiu** [comment]
Oh thanks for the catch @parasharrajat !

**Comment by michaelhaxhiu** [comment]
Blocked - hold.

**Comment by iwiznia** [comment]
Still on hold

**Comment by michaelhaxhiu** [comment]
Held still. Keeping assigned while I'm OOO for 1.5 weeks. If this becomes unblocked please re-assign a new BZ @iwiznia

**Comment by JmillsExpensify** [comment]
Firefox is not a supported browser. Closing
